module/VuFind/src/VuFind/Service/MarkdownFactory.php

```diff
@@ -135,7 +143,12 @@ protected function getEnvironment(): EnvironmentBuilderInterface
         $environment->addExtension(new CommonMarkCoreExtension());
         foreach ($this->extensions as $extension) {
             $extensionClass = $this->getExtensionClass($extension);
-            $environment->addExtension(new $extensionClass());
+            // For case, somebody needs to create extension using custom factory, we
+            // try to get the object from DI container if possible
+            $extensionObject = $this->container->has($extensionClass)
+                ? $this->container->get($extensionClass)
```

Comment:
Does it make sense to put extensions in the top-level container, or should we create a specialized container for them? (I'm not necessarily saying we should change anything -- just wondering if you've considered both possibilities).

Reply:
I don't expect too many custom extension to be added in typical instalation (if even any) so I think it would be just unnecessary overhead.